### ISSUE TITLE
Adding observability to ratelimiter

### ISSUE THREAD

**Issue by zhuangel** [cloud-hypervisor/cloud-hypervisor#5479]
**Describe the bug**
Cloud-hypervisor support software ratelimiter for virtio block and virtio net devices, now support configurations like, token bucket size, refill time, and one time burst token bucket size (initial burst size).

But the ratelimiter can not be monitored, we didn't know whether the ratelimiter works as expect, such as we could limit the IO into specified speed be setting token bucket size and refill time pair. But we don't know whether the limit happened, and unexpected limit happens because of lower token bucket size than IO burst size, or the IO limit contribute latencies to IO operations, even incorrect ratelimiter configuration is set.

We need to add some observability about ratelimiter, like counter for ratelimit happens, or some log for ratelimit happens (maybe one time log is enough).

**Comment by likebreath** [comment]
I agree that the behavior of rate-limiter can be hard to reason from its user configurations. Its behavior depends a lot on its internal implementations, user configurations, and workload being running. 

As mentioned in our I/O throttling documentation [1], one caveat when using rate-limiter is that every-time the bucket gets empty, it will stop I/O operations for a fixed amount of time (`cool_down_time`). The `cool_down_time` now is fixed at 100 ms, it can have big implications to the actual rate limit (which can be a lot different the expected "refill-rate" derived from user inputs). More details can be found from our documentation [1].

Our `rate-limiter` crate was borrowed from Firecracker [2] and has not been updated since it was introduced (#2333). I would think it is a good idea to check with the FC community first to get inputs on what's their practice of tuning rate-limiter. Also, it is a good idea to check what's the latest `rate-limiter` crate provides and see if any related improvements are available.

[1] https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/io_throttling.md
[2] https://github.com/firecracker-microvm/firecracker/tree/main/src/rate_limiter

**Comment by rbradford** [comment]
I think it is probably a reasonable to have some counters for when the rate limiter fires and rejects some ops/data.

e.g. `rate_limiter_exceeded_bytes` and `rate_limiter_exceeded_ops` - whether they can help with tuning the rate limiter is less clear to me but they could be useful to indicate that a workload is triggering the rate limiter.
